tests/unit/test-cmd/test.build.js

```diff
@@ -64,313 +56,269 @@ describe('build', () => {
-          args.shouldWatchFile('/some/path');
-          sinon.assert.called(fileFilter.wantFile);
-
-          // Remove the built extension.
-          return fs.unlink(buildResult.extensionPath)
-            // Execute the onChange handler to make sure it gets built
-            // again. This simulates what happens when the file watcher
-            // executes the callback.
-            .then(() => args.onChange());
-        })
-        .then((buildResult) => {
-          assert.match(buildResult.extensionPath, /\.zip$/);
-          return fs.stat(buildResult.extensionPath);
-        })
-        .then((stat) => {
-          assert.equal(stat.isFile(), true);
-        });
+      });
+      // Make sure we still have a build result.
+      assert.match(buildResult.extensionPath, /\.zip$/);
+      assert.equal(onSourceChange.called, true);
+      const args = onSourceChange.firstCall.args[0];
+      assert.equal(args.sourceDir, sourceDir);
+      assert.equal(args.artifactsDir, artifactsDir);
+      assert.typeOf(args.onChange, 'function');
+
+      // Make sure it uses the file filter.
+      assert.typeOf(args.shouldWatchFile, 'function');
+      args.shouldWatchFile('/some/path');
+      assert.equal(fileFilter.wantFile.called, true);
+
+      // Remove the built extension.
+      await fs.unlink(buildResult.extensionPath);
+      // Execute the onChange handler to make sure it gets built
+      // again. This simulates what happens when the file watcher
+      // executes the callback.
+      await args.onChange();
+      assert.match(buildResult.extensionPath, /\.zip$/);
+      const stat = await fs.stat(buildResult.extensionPath);
+      assert.equal(stat.isFile(), true);
     }
   ));
 
   it('throws errors when rebuilding in source watcher', () => withTempDir(
-    (tmpDir) => {
+    async (tmpDir) => {
       var packageResult = Promise.resolve({});
       const packageCreator = sinon.spy(() => packageResult);
       const onSourceChange = sinon.spy(() => {});
-      return build({
+      await build({
         sourceDir: fixturePath('minimal-web-ext'),
         artifactsDir: tmpDir.path(),
         asNeeded: true,
       }, {
         manifestData: basicManifest, onSourceChange, packageCreator,
-      })
-        .then(() => {
-          sinon.assert.called(onSourceChange);
-          sinon.assert.calledOnce(packageCreator);
-          const {onChange} = onSourceChange.firstCall.args[0];
-          packageResult = Promise.reject(new Error(
-            'Simulate an error on the second call to packageCreator()'));
-          // Invoke the stub packageCreator() again which should throw an error
-          return onChange();
-        })
-        .then(makeSureItFails())
-        .catch((error) => {
-          assert.include(
-            error.message,
-            'Simulate an error on the second call to packageCreator()');
-        });
+      });
+      assert.equal(onSourceChange.called, true);
```

Comment:
Let's use the sinon assetions in this test (as in the previous version of the same test).